backend/app/analysis/text_to_music.py

```diff
@@ -0,0 +1,179 @@
+import string
+from nltk.sentiment.vader import SentimentIntensityAnalyzer
+import numpy as np
+from scipy.io import wavfile
+import io
+import base64
+from nltk.tokenize import sent_tokenize
+import random
+
+
+musical_chars = {'a', 'b', 'c', 'd', 'e', 'f', 'g'}
+mc_list = list(musical_chars)
+
+# frequencies found from https://pages.mtu.edu/~suits/notefreqs.html
+note_freqs = {
+    'a': 440,
+    'b': 494,
+    'c': 523,
+    'd': 587,
+    'e': 330,
+    'f': 349,
+    'g': 392
+}
+
+# Ratios found from Wikipedia
+# (https://en.wikipedia.org/wiki/Consonance_and_dissonance#/media/File:Dyadic_harmonic_entropy_graph_(optimized_for_low_resolution).png)
+dissonant_ratios = [(5, 6), (4, 7), (5, 8), (5, 7), (6, 7)]
+neutral_ratios = [(3, 4), (3, 5), (4, 5)]
+consonant_ratios = [(1, 2), (2, 3)]
+
+
+def analyse_sentiment(text):
+    """
+    :param text: a string of text
+    :return: analysis of the text
+    """
+    # Convert text to lowercase
+    lower_case = text.lower()
+    # Removing punctuations
+    cleaned_text = lower_case.translate(str.maketrans('', '', string.punctuation))
+    score = SentimentIntensityAnalyzer().polarity_scores(cleaned_text)
+    return score
+
+
+def text_to_note(text):
+    """
+    :param text: Takes in a string of text
+    :return: A sound based on how negative or positive the text is.
+    """
+    score = analyse_sentiment(text)
+    note_freq = 400 + (score["pos"] - score["neg"]) * 350 * (1 + score["neu"])
```

Comment:
I might recommend pulling line 51 here out into a separate, 'private' function (`_generate_note_frequency`, vel sim.). In that function, you can use variable declarations to 'label' each of these what look like arbitrary integer values (400, 350, 1, etc.). While I trust that the implementation here makes sense, it'd be easier to parse in the future if you could examine another function that looked something like this:

```
def _generate_note_frequency(score_positive, score_negative, score_neutral):
  base_frequency = 400
  rounding_frequency = 350
  base_percentage = 1
  positivity_differential = score_positive = score_negative
  rounded_neutral_score = base_percentage + score_neutral
  return base_frequency + positivity_differential * rounding_frequency + rounded_neutral_score
```